### PR TITLE
subgraph docs

### PR DESCRIPTION
**Pull request by chidubemokeke**
Detailed subgraph documentation that defines the entities, explains how to query the subgraph, and lists examples of queries. This is helpful for any data consumer that intends to use your subgraph

### CODE REVIEW

File: docs/develop/data/subgraph-data.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+---
+sidebar_position: 1
+title: Subgraph Data Introduction
+---
+
+# Polygon Root Subgraph Introduction
+
+Polygon has a GraphQL API Endpoint hosted by [The Graph](https://thegraph.com/docs/about/introduction#what-the-graph-is) called a subgraph for indexing and organizing data from the Polygon smart contracts.
+
+This subgraph is can be used to query Polygon data.
```
Comment (DannyS03) [inline docs/develop/data/subgraph-data.md:10]:
```suggestion
This subgraph can be used to query on-chain Polygon data. The subgraph data is serviced by a decentralized group of server operators called [Indexers](https://thegraph.com/docs/en/network/indexing/).

```

File: docs/develop/data/subgraph-data.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+Subgraph information is serviced by a decentralized group of server operators called Indexers.
```
Comment (DannyS03) [inline docs/develop/data/subgraph-data.md:12]:
```suggestion
:::tip Get started
Learn more about how subgraphs work by checking out [The Graph's official documentation](https://thegraph.com/docs/en/). 
:::
```

File: docs/develop/data/subgraph-data.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+## Ethereum Mainnet
```
Comment (DannyS03) [inline docs/develop/data/subgraph-data.md:14]:
```suggestion
## Helpful resources
```

File: docs/develop/data/subgraph-data.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+[Creating an API Key Video Tutorial](https://www.youtube.com/watch?v=UrfIpm-Vlgs)
+
+- [Explorer Page](https://thegraph.com/explorer/subgraph?id=FDrqtqbp8LhG1hSnwtWB2hE6C97FWA54irrozjb2TtMH&view=Overview)
+- Graphql Endpoint: https://gateway.thegraph.com/api/[api-key]/subgraphs/id/FDrqtqbp8LhG1hSnwtWB2hE6C97FWA54irrozjb2TtMH
+- [Code Repo](https://github.com/maticnetwork/subgraphs)
+
+## Helpful Links
```
Comment (DannyS03) [inline docs/develop/data/subgraph-data.md:22]:
```suggestion
```

File: docs/develop/data/subgraph-data.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
```
Comment (DannyS03) [inline docs/develop/data/subgraph-data.md:23]:
```suggestion
```

File: docs/develop/data/subgraph-data.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+[Querying from an Application](https://thegraph.com/docs/en/developer/querying-from-your-app/)
```
Comment (DannyS03) [inline docs/develop/data/subgraph-data.md:24]:
```suggestion
- [Querying from an application](https://thegraph.com/docs/en/developer/querying-from-your-app/)
```

File: docs/develop/data/subgraph-data.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
```
Comment (DannyS03) [inline docs/develop/data/subgraph-data.md:25]:
```suggestion
```

File: docs/develop/data/subgraph-data.md [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+[Managing your API Key & Setting your indexer preferences](https://thegraph.com/docs/en/studio/managing-api-keys/)
```
Comment (DannyS03) [inline docs/develop/data/subgraph-data.md:26]:
```suggestion
- [Managing your API Key & setting your indexer preferences](https://thegraph.com/docs/en/studio/managing-api-keys/)
```

File: docs/develop/data/subgraph-data.md [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+Subgraph information is serviced by a decentralized group of server operators called Indexers.
+
+## Ethereum Mainnet
+
+[Creating an API Key Video Tutorial](https://www.youtube.com/watch?v=UrfIpm-Vlgs)
+
+- [Explorer Page](https://thegraph.com/explorer/subgraph?id=FDrqtqbp8LhG1hSnwtWB2hE6C97FWA54irrozjb2TtMH&view=Overview)
+- Graphql Endpoint: https://gateway.thegraph.com/api/[api-key]/subgraphs/id/FDrqtqbp8LhG1hSnwtWB2hE6C97FWA54irrozjb2TtMH
+- [Code Repo](https://github.com/maticnetwork/subgraphs)
+
```
Comment (DannyS03) [inline docs/develop/data/subgraph-data.md:21]:
```suggestion
```

File: docs/develop/data/subgraph-data.md [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+Polygon has a GraphQL API Endpoint hosted by [The Graph](https://thegraph.com/docs/about/introduction#what-the-graph-is) called a subgraph for indexing and organizing data from the Polygon smart contracts.
+
+This subgraph is can be used to query Polygon data.
+
+Subgraph information is serviced by a decentralized group of server operators called Indexers.
+
+## Ethereum Mainnet
+
+[Creating an API Key Video Tutorial](https://www.youtube.com/watch?v=UrfIpm-Vlgs)
+
```
Comment (DannyS03) [inline docs/develop/data/subgraph-data.md:17]:
```suggestion
```

File: docs/develop/data/subgraph-data.md [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+Polygon has a GraphQL API Endpoint hosted by [The Graph](https://thegraph.com/docs/about/introduction#what-the-graph-is) called a subgraph for indexing and organizing data from the Polygon smart contracts.
+
+This subgraph is can be used to query Polygon data.
+
+Subgraph information is serviced by a decentralized group of server operators called Indexers.
+
+## Ethereum Mainnet
+
+[Creating an API Key Video Tutorial](https://www.youtube.com/watch?v=UrfIpm-Vlgs)
```
Comment (DannyS03) [inline docs/develop/data/subgraph-data.md:16]:
```suggestion
[Video tutorial on creating an API Key](https://www.youtube.com/watch?v=UrfIpm-Vlgs)
```

**Review by DannyS03 — CHANGES_REQUESTED** [review]
Thank you for the contribution @chidubemokeke! 

- The descriptions for the entires seem to be missing.
- Can we also an additional note on how to use the explorer and the playground?
- Let's also change the metadata to follow [this format](https://wiki.polygon.technology/docs/contribute/writing-style#static-site-generator).

**Review by DannyS03 — CHANGES_REQUESTED** [review]
thanks @chidubemokeke! seems like we are only missing the reference in `sidebars.js` + a modification to the `id` name. For *entities.md*, it can be something like `id: graph-entities`, for queries: `id: graph-queries`, data: `graph-data`.

We may have forgotten to mention this in the contributor guide, but follows [this](https://docusaurus.io/docs/sidebar).

File: docs/develop/data/queries.md [hunk 12]
```diff
@@ -0,0 +1,65 @@
+---
+sidebar_position: 3
+title: Sample Queries
+---
+
+## Sample Queries
+
+Below are some sampl queries you can use to gather information from polygon root subgraph
```
[the remark this answers is not in the document]
Comment (chidubemokeke) [reply docs/develop/data/queries.md]: Done

File: docs/develop/data/subgraph-data.md (hunk 2 above)
Comment (chidubemokeke) [reply docs/develop/data/subgraph-data.md:25]: Done

File: docs/develop/data/subgraph-data.md (hunk 4 above)
Comment (chidubemokeke) [reply docs/develop/data/subgraph-data.md]: Done

File: docs/develop/data/graph-data.md [hunk 13]
```diff
@@ -0,0 +1,35 @@
+---
+id: Subgraph data
```
Comment (DannyS03) [inline docs/develop/data/graph-data.md:2]:
```suggestion
id: graph-data
```

File: docs/develop/data/graph-entities.md [hunk 14]
```diff
@@ -0,0 +1,328 @@
+---
+id: Subgraph entities
```
Comment (DannyS03) [inline docs/develop/data/graph-entities.md:2]:
```suggestion
id: graph-entities
```

File: docs/develop/data/graph-queries.md [hunk 15]
```diff
@@ -0,0 +1,78 @@
+---
+id: Query subgraphs
```
Comment (DannyS03) [inline docs/develop/data/graph-queries.md:2]:
```suggestion
id: graph-queries
```

**Review by DannyS03 — APPROVED** [review]
thanks again @chidubemokeke! Just changed the file ids which didn't match the ids in `sidebars.js`.

**Pull request merged**